src/reformat.c

```diff
@@ -44,6 +44,9 @@ avifBool avifPrepareReformatState(const avifImage * image, const avifRGBImage *
     state->mode = AVIF_REFORMAT_MODE_YUV_COEFFICIENTS;
 
     if (image->matrixCoefficients == AVIF_MATRIX_COEFFICIENTS_IDENTITY) {
+        if (image->yuvFormat != AVIF_PIXEL_FORMAT_YUV444) {
+            return AVIF_FALSE;
+        }
```

Comment:
Nit/optional: I suggest we check for this invalid combination earlier, before doing any work, i.e., before line 42.